src/status_im/contexts/wallet/swap/setup_swap/view.cljs

```diff
@@ -35,13 +36,20 @@
   (rf/dispatch [:wallet/clean-swap-proposal {:clean-approval-transaction? true}])
   (events-helper/navigate-back))
 
+(defn- start-get-swap-proposal
+  [amount clean-approval-transaction?]
+  (rf/dispatch [:wallet/stop-get-swap-proposal])
+  (rf/dispatch [:wallet.swap/set-loading-swap-proposal])
+  (debounce/debounce-and-dispatch [:wallet/start-get-swap-proposal
+                                   {:amount-in                   amount
+                                    :clean-approval-transaction? clean-approval-transaction?}]
+                                  swap-proposal-debounce-time-ms))
+
```

Comment:
We immediately push the loading state so the user can see instant feedback, and debounce the endpoint call.